rir/src/compiler/debugging/stream_logger.h

```diff
@@ -0,0 +1,63 @@
+#ifndef PIR_LOGGER_H
+#define PIR_LOGGER_H
+
+#include "../../runtime/Function.h"
+#include "../pir/pir_impl.h"
+#include "debugging.h"
+//#include "../translations/pir_2_rir.h"
+
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <stack>
+//#include <fstream>
+
+namespace rir {
+namespace pir {
+
+class StreamLogger {
+  public:
+    StreamLogger(DebugOptions options) : options(options) {}
+    ~StreamLogger() {
+        for (std::map<rir::Function*, std::stringstream*>::iterator it =
+                 streams.begin();
+             it != streams.end(); ++it) {
+            this->finish(it->first, *streams.at(it->first));
```

Comment:
can you manage to do this, when the function is done compiling and not only at the end? Because imagine you are in the middle of the compiler, and you want to see the inner function results as they are being compiled...

Reply:
Yes, maybe. The problem is where? Because if for instance inner functions do not run pir. So we would need a special flag for inner functions. Do you think is worth adding?